src/Parallelotope.jl

```diff
@@ -0,0 +1,125 @@
+export Parallelotope
+
+"""
+    Parallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}}
+     <: AbstractZonotope{N}
+
+Type that represents a parallelotope.
+
+### Fields
+
+- `offset`     -- vector; each element is the offset of the corresponding constraint
+- `directions` -- matrix ``D``; each row is the direction of two corresponding
+                  constraints
+
+### Notes
+``D_{i} = -D_{i+n}`` for ``i = 1, ..., n`` with ``n`` rows in ``D``
+
+"""
+struct Parallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}} <: AbstractZonotope{N}
+    offset::VN
+    directions::MN
+    center::VN
+
+    # default constructor with dimension check
+    function Parallelotope{N, VN, MN}(o::VN, D::MN, c::VN) where {N<:Real,
+         VN<:AbstractVector{N}, MN<:AbstractMatrix{N}}
+        @assert length(c) == checksquare(D)
+        return new{N, VN, MN}(o, D, c)
+    end
+end
+
+# convenience constructor without type parameter
+Parallelotope(o::VN, D::MN, c::VN) where {N<:Real,
```

Comment:
I think these constructors are a relic of previous Julia versions in this repository. You should be able to remove the type parameters from the inner constructor and then remove this constructor here.

Reply:
You can also keep them for now since you already wrote it. I created #1657 to remove them later again if we want to.